src/main/kotlin/com/mdrobnak/lalrpop/inspections/PrecedenceInspection.kt

```diff
@@ -1,74 +1,82 @@
-                        hasPrecedence = true
-                        val arg = precedenceAnnotation.annotationArg
+            precedenceAnnotationNullable?.let { precedenceAnnotation ->
+                val arg = precedenceAnnotation.annotationArg
 
-                        if (arg == null || arg.annotationArgName.text != "level") {
-                            holder.registerProblem(precedenceAnnotation, "Missing `level` arg on #[precedence]")
-                        } else {
-                            val levelText = arg.annotationArgValue.text
+                if (arg == null || arg.annotationArgName.text != "level") {
+                    holder.registerProblem(precedenceAnnotation, "Missing `level` arg on #[precedence]")
+                } else {
+                    val levelText = arg.annotationArgValue.text
 
-                            val level = levelText?.substring(1, levelText.length - 1) // remove the "
-                                ?.toIntOrNull()
+                    val level = levelText.substring(1, levelText.length - 1) // remove the "
+                        .toIntOrNull()
 
-                            if (level == null) {
-                                holder.registerProblem(arg.annotationArgValue, "Level value is not an int")
-                            }
-
-                            if (level == 1) {
-                                if (assocAnnotation != null) {
-                                    holder.registerProblem(
-                                        assocAnnotation,
-                                        "#[assoc] used on alternative with #[precedence(level=\"1\")]"
-                                    )
-                                }
-                            }
-                        }
+                    if (level == null) {
+                        holder.registerProblem(arg.annotationArgValue, "Level value is not an int")
                     }
 
-
-                    if (assocAnnotation != null) {
-                        val assocAnnotationArg = assocAnnotation.annotationArg
-
-                        if (assocAnnotationArg == null || assocAnnotationArg.annotationArgName.text != "side") {
-                            holder.registerProblem(assocAnnotation, "Missing side=\"...\" on #[assoc] annotation")
-                        }
+                    if (level == 1 && assocAnnotation != null) {
+                        holder.registerProblem(
+                            assocAnnotation,
+                            """#[assoc] used on alternative with #[precedence(level="1")]"""
+                        )
                     }
+                }
+            }
 
-                    if (precedenceAnnotation == null) {
-                        alternativesWithoutPrecedenceAnnotation.add(it)
+            if (assocAnnotation != null) {
+                assocAnnotation.annotationArg.takeUnless { it != null && it.annotationArgName.text == "side" }
```

Comment:
I would switch this to `takeIf`:
```suggestion
                assocAnnotation.annotationArg.takeIf { it == null || it.annotationArgName.text != "side" }
```